### PR TITLE
GH-92955: fix memory leak in code object lines and positions iterators

### LINKED ISSUE #92955: Memory leak in code object lines and positions iterator
The static types `LineIterator` and `PositionsIterator`  in `codeobject.c` are not finalized at exit leading to memory leak.

Reproducer:

```py
def func() -> None:
    pass

print(type(func.__code__.co_positions()).mro())
print(type(func.__code__.co_lines()).mro())
```

Memory leak:
```console
[<class 'positions_iterator'>, <class 'object'>]
[<class 'line_iterator'>, <class 'object'>]
[138 refs, 54 blocks]
```

### PR DESCRIPTION
**Pull request by kumaraditya303**
Closes #92955

### CODE REVIEW

**Review by corona10 — COMMENTED** [review]
@vstinner @erlend-aasland 
I am not sure that we need to handle all these cases, 
or is it better to wait for approval for converting static types into heap types from SC

**Review by corona10 — APPROVED** [review]
Since this will be the last change under Objects/*.c, I am happy to approve this change.

**Comment by kumaraditya303** [comment]
Thanks @corona10

**Pull request merged**
